### ISSUE TITLE
Invalid JSON response from server causes action to fail with no details

### ISSUE THREAD

**Issue by bradegler** [abcxyz/github-token-minter#42]
### TL;DR

There is a situation where the GitHub installation/access_token call returns a 201 even though it doesn't respond with a valid JSON token.

That response gets sent back to the action which tries to parse it and fails. 

### Expected behavior

Expected the github-token-minter server to respond with an error response instead of allowing invalid JSON to propagate to the client (GitHub action).

Expected the GitHub action to log more detailed information in this situation where the server responded with something other than JSON.

### Observed behavior

Action failed with an error that was not actionable and left no messages in the logs.

`Error: FetchError: invalid json response body at <prod url> reason: Unexpected end of JSON input`

### Environment Details

```markdown
GitHub Token Minter Production
```

### Additional information

NOTE: It would be nice if GitHub didn't respond with a 201 when it didn't actually produce a token.
